src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -317,6 +322,64 @@ public Response replaceFileInDataset(
         }
             
     } // end: replaceFileInDataset
+
+    /**
+     * Delete an Existing File 
+     * 
+     * @param id file ID or peristent ID
+     */
+    @DELETE
+    @Path("{id}")
+    public Response deleteFileInDataset(@PathParam("id") String fileIdOrPersistentId){
+
+        if (!systemConfig.isHTTPUpload()) {
```

Comment:
Why is this isHTTPUpload check here? Is it necessary? Here's what's in the bundle, which seems to be about upload:

file.api.httpDisabled=File upload via HTTP is not available for this installation of Dataverse.

I guess the feeling is that if upload is disabled, delete should be disabled too? However, I don't see this check in the SWORD code for "delete file": https://github.com/IQSS/dataverse/blob/v5.13/src/main/java/edu/harvard/iq/dataverse/api/datadeposit/MediaResourceManagerImpl.java#L146

Reply:
I have removed the check. If we ever decide that it is needed, we can add it back.